File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/MgMlConcentration.java
package com.labessence.biotech.finaldilution.component.concentration;

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

public class MgMlConcentration extends Concentration {
    public MgMlConcentration(double amount) {
        super(amount, ConcentrationType.MILIGRAM_PER_MILLILITER);
    }

    @Override
    public double calcDesiredMass(double volume, double molarMass) {
        return concentration * volume / 1000;
    }

    @Override
    public double calcVolumeForDesiredMass(double mass, double molarMass) {
        return mass / concentration * 1000;
    }
}

File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/MgMlConcentration.kt
package com.labessence.biotech.finaldilution.component.concentration

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

class MgMlConcentration(amount: Double) : Concentration(amount, ConcentrationType.MILIGRAM_PER_MILLILITER) {

    override fun calcDesiredMass(volume: Double, molarMass: Double): Double {
        return concentration * volume / 1000
    }

    override fun calcVolumeForDesiredMass(mass: Double, molarMass: Double): Double {
        return mass / concentration * 1000
    }
}
